### PR TITLE
Add logout link

### PR DESCRIPTION
**Pull request by erich**
Remove unused function from AuthController
Add AuthControllerTest
Add signed_in? function

image

### CODE REVIEW

File: lib/reddex_web/plugs/auth.ex [hunk 1]
```diff
@@ -8,7 +8,7 @@ defmodule ReddexWeb.Plugs.Auth do
   def call(conn, _) do
     # TODO: Not sure about this
     # maybe be more explicit in controllers with actions
-    if get_session(conn, :current_user) || conn.request_path == "/sign_in" do
+    if get_session(conn, :current_user) || conn.request_path == "/sign_in" || conn.request_path == "/logout" do
```
Comment (Math3v) [inline lib/reddex_web/plugs/auth.ex]: This is getting weird. How about something like this?

```
if get_session(conn, :current_user) || public_path?(conn.request_path) do
# ...
end

defp public_path?("/sign_in"), do: true
defp public_path?("/logout"), do: true
defp public_path?(_), do: false
```

File: test/reddex_web/controllers/auth_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  setup %{conn: conn} do
+    user = %Reddex.Accounts.User{name: "dhh"}
+    conn = Plug.Test.init_test_session(conn, current_user: user)
+    {:ok, conn: conn}
+  end
+
+  describe "delete" do
+    test "logs out an user", %{conn: conn} do
+      conn = delete(conn, auth_path(conn, :delete))
+      assert redirected_to(conn) == "/sign_in"
```
Comment (Math3v) [inline test/reddex_web/controllers/auth_controller_test.exs]: assert for `:current_user`?

File: lib/reddex_web/views/layout_view.ex [hunk 3]
```diff
@@ -1,3 +1,7 @@
 defmodule ReddexWeb.LayoutView do
   use ReddexWeb, :view
+
+  def signed_in?(conn) do
+    Plug.Conn.get_session(conn, :current_user)
```
Comment (Math3v) [inline lib/reddex_web/views/layout_view.ex]: `!!Plug.Conn.get_session(conn, :current_user)` ?

I'd expect the `signed_in?` function to return `true/false`.

**Comment by erich** [comment]
@Math3v thanks, pushed

**Comment by Math3v** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 fixed issue! 🎉

But beware that this branch is **[5 commits behind the `PrimeHammer:master` branch](https://github.com/PrimeHammer/reddex/compare/PrimeHammer:logout...PrimeHammer:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/PrimeHammer/reddex/pulls/21.

**Comment by Math3v** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/PrimeHammer/reddex/pulls/21.

**Pull request merged**